### PR TITLE
[Mellanox] Remove unsupported air flow direction from test_dynamic_minimum_table

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**
### Description of PR

Summary:
Remove unsupported fan direction from test_dynamic_minimum_table. And make test_set_psu_fan_speed more robust.

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?

1. It is hard to determine air flow direction at run time, so the current thermal control algorithm assumes that air flow direction is unknown. Remove unsupported air flow direction from test_dynamic_minimum_table. 
2. The fan speed might be adjusted by kernel. Make test_set_psu_fan_speed more robust. 

#### How did you do it?

1. Remove unsupported air flow direction check
2. When fan speed keep 100% after making all fan in normal state, skip test_set_psu_fan_speed

#### How did you verify/test it?

Manually run regression case

#### Any platform specific information?

Mellanox

#### Supported testbed topology if it's a new test case?

N/A

### CODE REVIEW

**Comment by Junchao-Mellanox** [comment]
retest vsimage please

**Review by keboliu — APPROVED** [review]

File: tests/platform_tests/mellanox/test_thermal_control.py [hunk 1]
```diff
@@ -89,11 +81,31 @@ def test_set_psu_fan_speed(duthosts, rand_one_dut_hostname, mocker_factory):
 
     logging.info('Mock FAN presence...')
     single_fan_mocker.mock_presence()
-    assert wait_until(THERMAL_CONTROL_TEST_WAIT_TIME, THERMAL_CONTROL_TEST_CHECK_INTERVAL, check_cooling_cur_state, duthost, 10, operator.ne), \
-        'Current cooling state is {}'.format(get_cooling_cur_state(duthost))
-    logging.info('Wait {} seconds for the policy to take effect...'.format(THERMAL_CONTROL_TEST_CHECK_INTERVAL))
-    time.sleep(THERMAL_CONTROL_TEST_CHECK_INTERVAL)
+    wait_until(THERMAL_CONTROL_TEST_WAIT_TIME, THERMAL_CONTROL_TEST_CHECK_INTERVAL, check_cooling_cur_state, duthost, 10, operator.ne)
+    logging.info('Wait {} seconds for the policy to take effect...'.format(THERMAL_CONTROL_TEST_WAIT_TIME * 2))
+    # We have to wait THERMAL_CONTROL_TEST_WAIT_TIME * 2 seconds long here because:
```
Comment (liat-grozovik) [inline tests/platform_tests/mellanox/test_thermal_control.py:86]: rephrase comments. 
why change for all systems if there is a system which behaves differently?
please suggest alternative how to get the valid numbers.

File: tests/platform_tests/mellanox/test_thermal_control.py [hunk 2]
```diff
@@ -89,11 +81,31 @@ def test_set_psu_fan_speed(duthosts, rand_one_dut_hostname, mocker_factory):
...
+    #     1. We observe that on lionfish, it might take about 30 seconds for PSU to adjust
+    #     its fan speed to target value
+    #     2. Issue 2255767. There is chance that kernel might incorrectly change cooling state back to 10 after
+    #     user space thermal control adjust it to dynamic minimum value. This issue is set to "Wont Fix"
+    #     by hw-mgmt team. So we have to wait longer for the user space thermal control
+    #     to set fan speed to dynamic minimum value again (1 more minutes).
+    time.sleep(THERMAL_CONTROL_TEST_WAIT_TIME * 2)
     cooling_cur_state = get_cooling_cur_state(duthost)
+    if cooling_cur_state == 10:
```
Comment (liat-grozovik) [inline tests/platform_tests/mellanox/test_thermal_control.py:95]: what does 10 means?

File: tests/platform_tests/mellanox/test_thermal_control.py [hunk 3]
```diff
@@ -4,17 +4,18 @@
...
+from tabulate import tabulate
+import re
 
 pytestmark = [
     pytest.mark.asic('mellanox'),
     pytest.mark.topology('any')
 ]
 
-THERMAL_CONTROL_TEST_WAIT_TIME = 65
+THERMAL_CONTROL_TEST_WAIT_TIME = 75
```
Comment (liat-grozovik) [inline tests/platform_tests/mellanox/test_thermal_control.py:18]: why more time is needed?

**Review by liat-grozovik — CHANGES_REQUESTED** [review]

File: tests/platform_tests/mellanox/test_thermal_control.py (hunk 3 above)
Comment (Junchao-Mellanox) [reply tests/platform_tests/mellanox/test_thermal_control.py:18]: Thermal loop usually takes 60 seconds, but we found some slow sysfs read in a few testbeds, in that case, wait 65 seconds is not enough. We talked to low level team, they claimed that the slow sysfs read is not an issue, so we have to enhance the wait time here.

File: tests/platform_tests/mellanox/test_thermal_control.py (hunk 2 above)
Comment (Junchao-Mellanox) [reply tests/platform_tests/mellanox/test_thermal_control.py:95]: It means max cooling level. I will change it to a constant.

File: tests/platform_tests/mellanox/test_thermal_control.py (hunk 1 above)
Comment (Junchao-Mellanox) [reply tests/platform_tests/mellanox/test_thermal_control.py:88]: Please see new comments.

**Review by liat-grozovik — APPROVED** [review]

**Review by wangxin — APPROVED** [review]

**Pull request merged**